src/osemgrep/core/Metrics_.ml

```diff
@@ -123,6 +137,49 @@ open Semgrep_metrics_t
 
 let add_engine_type ~name = g.payload.value.engineRequested <- name
 let is_using_registry () = g.is_using_registry
+let is_enabled () = g.config <> Off
+
+let set_is_using_registry ~is_using_registry =
+  g.is_using_registry <- is_using_registry
+
+let set_anonymous_user_id ~anonymous_user_id =
+  g.payload.anonymous_user_id <- anonymous_user_id
+
+let set_started_at ~started_at = g.payload.started_at <- started_at
```

Comment:
This is a bit annoying to have all those wrapper functions for each field ...
we should just have a payload() function that the caller just assign the field directly.

Reply:
I'm open to other implementations. 

In python we might do something like
```
>>> payload = {}
>>> payload.update({"x": 2, "y": 3})
>>> payload
{'x': 2, 'y': 3}
```

If you can point me to code reference or sample you have in mind I can take another pass.

Reply:
no we don't want that because we want proper typechecking (like using an undefined field).
But you can keep it like that; I'll refactor things in a separate PR.